src/adservice/src/main/java/hipstershop/AdService.java

```diff
@@ -207,16 +207,24 @@ public static void initStackdriver() {
     // Registers logging trace exporter.
     LoggingTraceExporter.register();
     long sleepTime = 10; /* seconds */
-    int maxAttempts = 3;
+    int maxAttempts = 5;
+    boolean statsExporterRegistered = false;
+    boolean traceExporterRegistered = false;
```

Comment:
As far as I can tell you don't use these variables anywhere. Why introduce them?

Reply:
oh I see the logic now. I would've preferred if each operation had its own retries like `retry(initTracing())` and `retry(initStats())` but this is OK.